### PR TITLE
react-react1-week2

### PR DESCRIPTION
**Pull request by Merhawi-gebreslasie**

### CODE REVIEW

File: react/react1/week2/homework/my-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+]
+function App() { 
+
+
+  
+  return (
+    <div className="App">
+      <Header header='Todo list' />
+      <TodoList 
+      Noitems='No items'
```
Comment (dsamourkasidis) [inline react/react1/week2/homework/my-app/src/App.js]: You are passing Noitems but it doesn't exist in you TodoList component! If you don't need it you should remove it.
Also, naming :)
Noitems -> noItems

File: react/react1/week2/homework/my-app/src/components/TodoList.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { TodoItem } from "./TodoItem";
+import { useState } from "react";
+import { AddTodoItem } from "./AddTodoItem";
+export const TodoList = ({ todos }) => {
+  const [items, setItems] = useState(todos);
+
+  
+  const addTodos = (newTodo) => {
```
Comment (dsamourkasidis) [inline react/react1/week2/homework/my-app/src/components/TodoList.js]: I suggest you use consistent names. For instance, in this function, you're adding a single todo but you've named it addTodos. A better option would simply be addTodo

File: react/react1/week2/homework/my-app/src/components/AddTodoItem.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+export const AddTodoItem = ({ addTodos, items }) => {
+  const newId = items.length;
```
Comment (dsamourkasidis) [inline react/react1/week2/homework/my-app/src/components/AddTodoItem.js]: Using the array's length as a new id can lead to some items having the same id. Consider the following case:
1. The array has 3 items.
2. You add a new 4th item so it gets the id 4.
3. You remove the 1st item, so the array has 3 items again.
4. You add a new 4th, so it also gets the id 4.

One way to solve this is to use 
```Math.floor(Math.random() * 10000)``` to generate a random number.

**Review by dsamourkasidis — COMMENTED** [review]
Good job 👍 
Generally everything looks good, I suggest you go through my comment and make some small fixes though :)
Also, as a general comment try to improve a bit the formatting of your code (indentations, new lines etc..). You can actually use VSCode's own built-in [formatter](https://mkyong.com/vscode/how-to-format-source-code-in-visual-studio-code-vscode/)
It'll make you code look nicer and easier to read.

File: react/react1/week2/homework/my-app/src/App.js (hunk 1 above)
Comment (Merhawi-gebreslasie) [reply react/react1/week2/homework/my-app/src/App.js]: Thanks 👍 .

File: react/react1/week2/homework/my-app/src/components/TodoList.js (hunk 2 above)
Comment (Merhawi-gebreslasie) [reply react/react1/week2/homework/my-app/src/components/TodoList.js]: Yes, thanks

File: react/react1/week2/homework/my-app/src/components/AddTodoItem.js (hunk 3 above)
Comment (Merhawi-gebreslasie) [reply react/react1/week2/homework/my-app/src/components/AddTodoItem.js]: Thanks that  is nice 👍 , what about  Math.floor(Math.random() * 10000) can it be unique.

**Pull request merged**